backup/crypto.go

```diff
@@ -4,11 +4,78 @@ import (
 	"crypto/aes"
 	"crypto/cipher"
 	"crypto/rand"
+	"encoding/json"
+	"fmt"
 	"io"
 
 	"github.com/pkg/errors"
+	"golang.org/x/crypto/nacl/box"
 )
 
+// WrappedKey is the JSON-encoded "wrapped key" that a backup is encrypted with.
+type WrappedKey struct {
+	Nonce        []byte
+	CipherText   []byte
+	SenderPubkey []byte
+}
+
+// wrapKey takes a public key, and an aes key to encrypt to it.
+// It returns a string suitable for passing to `unwrap` (it's json)
+func wrap(recipientPubkey *[32]byte, keyToWrap []byte) (wrapped []byte, err error) {
+	senderPubkey, privateKey, err := box.GenerateKey(rand.Reader)
```

Comment:
Looks like we're not storing the private key once we're done with a call to `wrap`. Is this key pair only used for signature/verification and therefore we only need pubkey (which I see we do attach to the `WrappedKey`)?

If so it might be worth to comment on that in the code.

Reply:
the design doc goes over the cryptographic choices, but I'll leave a comment here too.